src/dags/sql/sport.py

```diff
@@ -32,10 +29,18 @@
         ALTER TABLE {{ params.tablename }}_new ADD COLUMN IF NOT EXISTS "geometry"
                 geometry(POINT, 28992) null;
         UPDATE {{ params.tablename }}_new
-        SET geometry = ST_Transform(ST_SetSRID(ST_MakePoint(REPLACE(x, ',', '.')::double precision,
-                REPLACE(y, ',', '.')::double precision), 4326), 28992)
-        WHERE (x !~* '[a-z]+' AND y !~* '[a-z]+')
-        AND (length(x) > 0 AND length(y) > 0);
+        SET geometry =
+            CASE WHEN x > y THEN
+                ST_Transform(ST_SetSRID(ST_MakePoint(REPLACE(y::text, ',', '.')::double precision,
+                REPLACE(x::text, ',', '.')::double precision), 4326), 28992)
+            ELSE
+                ST_Transform(ST_SetSRID(ST_MakePoint(REPLACE(x::text, ',', '.')::double precision,
+                REPLACE(y::text, ',', '.')::double precision), 4326), 28992)
+            END
+        WHERE 1=1
+        AND CASE WHEN x::text !~* '[a-z]+' then true else false end = true
```

Comment:
Good point. Strangely the casting of the value for X (or Y in that matter) conflicts with the use of the regexp. Since the casting seems to be done after the regexp leading to an invalid method error. The use of the case when seems to prevent this and casts before applying the regexp. However, I will double check it just to be sure.